### PR TITLE
CATL-1134: Dont show Bulk Email

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR the Bulk Emails have been hidden from the Civicase.

## Before
2019-09-27 at 3 02 PM

## After
2019-09-27 at 3 02 PM

## Technical Details
From the following files/components, the Bulk Email activities are filtered
1. `activities-calendar.directive.js`
2. `activity-feed.directive.js `
3. `get-case-query-params.factory.js`
4. `dashboard-tab.directive.js`
5. `Getdayswithactivities.php`
6. `Getdetails.php`

### CODE REVIEW

File: api/v3/Case/Getdetails.php [hunk 1]
```diff
@@ -154,6 +154,7 @@ function civicrm_api3_case_getdetails(array $params) {
         'case_id' => ['IN' => $ids],
         'is_current_revision' => 1,
         'is_test' => 0,
+        'activity_type_id' => ['!=' => "Bulk Email"],
```
Comment (tunbola) [inline api/v3/Case/Getdetails.php]: Bulk email can be in single quotes

Comment (tunbola) [reply api/v3/Case/Getdetails.php]: Hope there are no other places in the custom API's where Activity.get is being called and we need to exclude the bulk email activity?

Comment (deb1990) [reply api/v3/Case/Getdetails.php]: Good call. Removed the Bulk Email from one more place.

File: CRM/Civicase/Activity/ContactActivitiesSelector.php [hunk 2]
```diff
@@ -55,25 +71,31 @@ public function getActivitiesForContactCount($params) {
...
    * @return array
+   *   Paramaters.
    */
-  private function getParamsWithoutOffsetsAndLimits($params) {
+  private function getParamsWithoutOffsetsAndLimits(array $params) {
     $options = CRM_Utils_Array::value('options', $params, []);
     $options['limit'] = 0;
     $options['offset'] = 0;
     $params['options'] = $options;
+    $params['activity_type_id'] = ['!=' => 'Bulk Email'];
```
Comment (tunbola) [inline CRM/Civicase/Activity/ContactActivitiesSelector.php]: We might not need to add this actually. I'm assuming that Params is from the API request, so the FE will already have that sent.

Also as the name of the function implies, it's basically to remove offset and limit so it won't be proper to add what is not related.

**Pull request merged**